lib/Page.js

```diff
@@ -226,28 +256,49 @@ Page.prototype.collectHeadingsInContent = function (content) {
         const includeContent = fs.readFileSync(includePath);
         if (panel.attribs.fragment) {
           $ = cheerio.load(includeContent);
-          this.collectHeadingsInContent($(`#${panel.attribs.fragment}`).html());
+          this.collectHeadingsAndKeywordsInContent($(`#${panel.attribs.fragment}`).html(), closestHeading);
         } else {
-          this.collectHeadingsInContent(includeContent);
+          this.collectHeadingsAndKeywordsInContent(includeContent, closestHeading);
         }
       } else {
-        this.collectHeadingsInContent($(panel).html());
+        this.collectHeadingsAndKeywordsInContent($(panel).html(), closestHeading);
       }
     });
   $ = cheerio.load(content);
   if (this.headingIndexingLevel > 0) {
-    let headingsSelector = 'h1';
-    for (let i = 2; i <= this.headingIndexingLevel; i += 1) {
-      headingsSelector += `, h${i}`;
-    }
     $('modal').remove();
     $('panel').remove();
     $(headingsSelector).each((i, heading) => {
       this.headings[$(heading).attr('id')] = $(heading).text();
     });
+    $('.keyword').each((i, keyword) => {
+      let closestHeading = getClosestHeading($, headingsSelector, keyword);
+      if (!closestHeading) {
+        if (!lastHeading) {
+          logger.warn(`Missing heading for keyword: ${$(keyword).text()}`);
+          return;
+        }
+        closestHeading = lastHeading;
+      }
+      const headingId = $(closestHeading).attr('id');
```

Comment:
Was wondering whether line 283 - 287 can be refactored to a different method, took me a while to understand what was going on (maybe `linkKeywordToHeading(keyword, heading)`)?